rivgraph/im_utils.py

```diff
@@ -1556,3 +1556,54 @@ def make_mask(Ishape, coords):
         I[Imask==True] = 1
 
     return I
+
+
+def grow_blob(I, n=1, strel='square'):
+    
+    """
+    Grow labelled blobs using a specified kernel shape.

```

Comment:
Can you add more detail here about what "growing a blob" is? It seems similar (the same?) as dilation. If that's the case, then we already have this functionality.

Reply:
So this is very very similar to grayscale dilation, but it has a step in it to check what happens when 2 dilating (growing) blobs begin to bump into another. Grayscale dilation will mix the values that are near the intersection. This is correct in the sense of grayscale dilation, but I have been experimenting with growing "blobs" for things like conditional dilation or growing lakes until they bump into one another. In that case, when two blobs bump into one another, we want them to not grow anymore.

To address this, we define a rule at the boundaries of two growing blobs: if two blobs X and Y bump into one another, i.e. if any additional pixels X' added after a dilation intersect any pixels which belonged to a different blob Y, the intersected pixels are set to the original value Y. In the case of two lakes which are dilating into the same position the pixels should consistently take either the higher or the lower object number (not sure exactly which in Python but it'll be internally consistent).

This doesn't necessarily need to be included, but it could eventually be part of some lake identification/removal functionality I've been playing with.

Reply:
Ok, I see. This has the potential to be really useful with a (major or minor, I'm not sure) change. In the mouse brain example, I try erosion + dilation to smooth the boundaries of the blobs, but we end up merging blobs even with the smallest kernel possible and one iteration. So I wonder if it would be possible to ensure that _disconnected blobs remain disconnected after dilation_, or another way to say that is you always have the same number of blobs post-dilation as you do in the original image. This may be very difficult to implement--I can't immediately tell.

Another point here is that the morphologic operators in RG are designed for use on binary images only, so there are no "higher and lower" object numbers. Perhaps we could relax this restriction if there are sufficient use cases. What is the use case you're envisioning that involves dilation of non-binary images?